pyuvdata/uvbeam/uvbeam.py

```diff
@@ -3820,6 +3846,18 @@ def read(
         fix_auto_power : bool
             For power beams, if auto polarization beams with imaginary values are found,
             fix those values so that they are real-only in data_array.
+        freq_range : tuple of float in Hz
+            If given, the lower and upper limit of the frequencies to read in. Default
+            is to read in all frequencies. Restricting the frequencies reduces peak
+            memory usage.
+        az_range : tuple of float in deg
+            The azimuth range to read in, if the beam is specified in az/za coordinates.
+            Default is to read in all azimuths. Restricting the azimuth reduces peak
+            memory usage.
+        za_range : tuple of float in deg
+            The zenith angle range to read in, if the beam is specified in za/za
+            coordinates. Default is to read in all za. Restricting the za reduces peak
+            memory.
```

Comment:
Please specify when these will be used (only for beam fits files, az/za only for az/za beam fits files).

Reply:
I am wondering whether we should implement these same parameters for the `select` method to keep the interfaces similar (as we do in UVData) or whether that doesn't matter.

Reply:
I removed the `freq_range` entry because it was duplicate. I added some info to the existing entry. Also added that the az/za works for beamfits az/za grid files. I don't know if we should add to `select`...